MarkBind/markbind
# [Do not merge] Fixed markbind convert extra newline issue

**What is the purpose of this pull request?**



Fixes #1726 
Fixes #1737 
Fixes #1736 
Fixes #1740 

**Overview of changes:**
This PR fixed the issues raised in #1726 and #1736.

This was caused by the extra newline created at the end of buildSiteNav() in the core/site dir. I simply added a trimEnd() to fix the issue.
While working on this issue, I ran into alot of trouble while adding new tests, as they kept failing the CI check, turns out the issue was that markbind convert was incorrectly generating the file paths as described in #1736.

After fixing this issue, `npm run updatetest` works and the test now passes.

**Anything you'd like to highlight / discuss:**
Since there are many file changes I have highlighted the files that should be looked at below.

General: 
All functional changes are made in index.js in `cli`

For newline change:
The Home.html in `test_navigation_convert` should not contain any unnecessary spaces in the navigation component. 

For backslash change:
The about.html in `test_navigfation_convert` should not have any instance of `\` or `%5C`

**Testing instructions:**

For newline change:
Run `markbind init -convert` on a test repo and see that there is no more new line before the closing </site-nav> tag.

For backslash change:
Run `markbind init -convert` on a test repo and make sure there are no instances of `\`


**Proposed commit message: (wrap lines at 72 characters)**

Fixed `markbind convert` extra newline
and backslash issues

---

**Checklist:** :ballot_box_with_check:
